helm/cluster-vsphere/templates/_helpers.tpl

```diff
@@ -219,3 +219,13 @@ To enforce upgrades, a version suffix is appended to secret name.
 {{- define "credentialSecretName" -}}
 {{- include "resource.default.name" $ }}-credentials
 {{- end -}}
+
+{{- define "imageRepo" -}}
```

Comment:
I wouldn't have this check here. If the user can set .controlPlane.image.repository as projects.registry.vmware.com/tkg for 1.22 in the values.yaml file.

Reply:
Is the repo to use based on the K8s version? Is this documented somewhere?

Reply:
```
d pull projects.registry.vmware.com/tkg/kube-apiserver:v1.22.5_vmware.1 OK
d pull projects.registry.vmware.com/tkg/kube-apiserver:v1.24.11_vmware.1 OK
d pull projects.registry.vmware.com/tkg/kube-apiserver:v1.25.6_vmware.1 NOT OK 
d pull projects.registry.vmware.com/tkg/kube-apiserver:v1.26.0_vmware.1 NOT OK
d pull projects.registry.vmware.com/tkg/kube-apiserver:v1.27.0_vmware.1 NOT OK
```

it looks like since 1.25 they started using the upstream images (https://kubernetes.slack.com/archives/CKFGK3SSD/p1680109338105649?thread_ts=1680095920.864939&cid=CKFGK3SSD). I don't know why this had to be there before, but for 1.24 the community image also works w/o any issues. I was assuming there are some reasons to have it there so I've added this check that's backward compatible and doesn't require the user of this api to remember what repo to use for what version of k8s. I mean yes, it's doable, but is it UX sucks. If I had to look up some table of versions -> image repos when working w/ the helm chart I wouldn't be very happy user :)

If we can ignore everything before 1.24 then other option would be to get rid of that `projects.registry.vmware.com/tkg` repo altogether. Are there any reason to use it for coredns image, or etcd image? Is that documented somewhere?

Reply:
I prefer starting with 1.24 and forgetting the old versions :)

Reply:
ok, makes sense